core/src/main/java/io/questdb/griffin/engine/table/ShowAllTablePartitionsCursoryFactory.java

```diff
@@ -0,0 +1,321 @@
+                                            if (detachedTxReader != null) {
+                                                detachedTxReader.clear();
+                                            }
+                                        }
+                                    } else {
+                                        LOG.error().$("detached partition does not have meta file [path=").$(path).I$();
+                                    }
+                                } else {
+                                    LOG.error().$("detached partition meta does not match [path=").$(path).I$();
+                                }
+                            } finally {
+                                if (detachedMetaReader != null) {
+                                    detachedMetaReader.clear();
+                                }
+                            }
+                        } else {
+                            LOG.error().$("detached partition does not have meta file [path=").$(path).I$();
+                        }
+                        path.parent();
+                    }
+                    sizeB += ff.getDirSize(path);
+                    partitionName.clear();
+                    partitionIndex++;
+                }
+            }
+
+            private void scanDetachedAndAttachablePartitions() {
+                long pFind = ff.findFirst(path.$());
+                if (pFind > 0L) {
+                    try {
+                        attachablePartitions.clear();
+                        detachedPartitions.clear();
+                        do {
+                            partitionName.clear();
+                            long name = ff.findName(pFind);
+                            Utf8s.utf8ToUtf16Z(name, partitionName);
+                            int type = ff.findType(pFind);
+                            if ((type == Files.DT_LNK || type == Files.DT_DIR) && Chars.endsWith(partitionName, TableUtils.ATTACHABLE_DIR_MARKER)) {
+                                attachablePartitions.add(Chars.toString(partitionName));
+                            } else if (type == Files.DT_DIR && CairoKeywords.isDetachedDirMarker(name)) {
+                                detachedPartitions.add(Chars.toString(partitionName));
+                            }
+                        } while (ff.findNext(pFind) > 0);
+                        attachablePartitions.sort(CHAR_COMPARATOR);
+                        detachedPartitions.sort(CHAR_COMPARATOR);
+                    } finally {
+                        ff.findClose(pFind);
+                    }
+                }
+            }
+        }
+    }
+
+    static {
+        TABLE_NAME = 0;
+        WAL_ENABLED = 1;
+        PARTITION_BY = 2;
+        PARTITION_COUNT = 3;
+        ROW_COUNT = 4;
+        SIZE_B = 5;
+
+        final GenericRecordMetadata metadata = new GenericRecordMetadata();
+        metadata.add(new TableColumnMetadata("table_name", ColumnType.STRING));
```

Comment:
Please align naming with talbe_partitions function, should be camel case:
- tableName
- walEnabled
- partitionBy
- partitionCount
- rowCount
- diskSize